elasticdl/python/allreduce/torch_optimizer.py

```diff
@@ -0,0 +1,294 @@
+# Copyright 2020 The ElasticDL Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import warnings
+from contextlib import contextmanager
+
+import torch
+from horovod.torch.compression import Compression
+from horovod.torch.mpi_ops import Average, allreduce_async_, size, synchronize
+
+
+class _ElasticDistributedOptimizer(torch.optim.Optimizer):
```

Comment:
Please add the permalink of horovod code snippet which we refer to.

Reply:
Good suggestion!